src/storage/hummock_sdk/src/compaction_group/mod.rs

```diff
@@ -154,4 +358,123 @@ pub mod group_split {
 
         Err(target_levels.len())
     }
+
+    /// Split the SSTs in the level according to the split key.
+    pub fn split_sst_info_for_level_v2(
+        level: &mut Level,
+        new_sst_id: &mut u64,
+        split_key: Bytes,
+    ) -> Vec<SstableInfo> {
+        if level.table_infos.is_empty() {
+            return vec![];
+        }
+
+        let mut insert_table_infos = vec![];
+        if level.level_type == PbLevelType::Overlapping {
+            level.table_infos.retain_mut(|sst| {
+                let sst_split_type = need_to_split(sst, split_key.clone());
+                match sst_split_type {
+                    SstSplitType::Left => true,
+                    SstSplitType::Right => {
+                        insert_table_infos.push(sst.clone());
+                        false
+                    }
+                    SstSplitType::Both => {
+                        let estimated_size = sst.sst_size;
+                        let branch_sst = split_sst(
+                            sst,
+                            new_sst_id,
+                            split_key.clone(),
+                            estimated_size / 2,
+                            estimated_size / 2,
+                        );
+                        insert_table_infos.push(branch_sst.clone());
+                        true
+                    }
+                }
+            });
+        } else {
+            let pos = get_split_pos(&level.table_infos, split_key.clone());
+            if pos >= level.table_infos.len() {
+                return insert_table_infos;
+            }
+
+            let sst = &mut level.table_infos[pos];
+            let sst_split_type = need_to_split(sst, split_key.clone());
+            match sst_split_type {
+                SstSplitType::Left => {
+                    insert_table_infos.extend_from_slice(&level.table_infos[pos + 1..]);
+                    level.table_infos = level.table_infos[0..=pos].to_vec();
+                }
+                SstSplitType::Right => {
+                    insert_table_infos.extend_from_slice(&level.table_infos[pos..]); // the sst at pos has been split to the right
```

Comment:
If we reach here, pos must be 0? Can we add an assertion for this?

Reply:
I don't fully understand what you mean, L409's SstSplitType corresponds to the type of the sst at the pos. 

The value of pos depends on the values of table_infos and split_key in L397, maybe I misunderstood you?

Reply:
We use `get_split_pos` to find the last SST with `key_range_left < split_key` if any or return pos 0. If `need_to_split` returns `SstSplitType::Right`, that means `split_key >= key_range_left`. The only possibilities here is pos==0.

Reply:
Got it, add assertion